source/Octopus.Tentacle.Tests.Integration/ClientFileTransfersAreNotRetriedWhenRetriesAreDisabled.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Halibut;
+using NUnit.Framework;
+using Octopus.Tentacle.Tests.Integration.Support;
+using Octopus.Tentacle.Tests.Integration.Util.Builders;
+using Octopus.Tentacle.Tests.Integration.Util.Builders.Decorators;
+using Octopus.Tentacle.Tests.Integration.Util.TcpTentacleHelpers;
+
+namespace Octopus.Tentacle.Tests.Integration
+{
+    [IntegrationTestTimeout]
+    public class ClientFileTransfersAreNotRetriedWhenRetriesAreDisabled : IntegrationTest
+    {
+        [Test]
+        [TestCaseSource(typeof(TentacleTypesAndCommonVersionsToTest))]
+        public async Task FailedUploadsAreNotRetriedAndFail(TentacleType tentacleType, string version)
+        {
+            using var clientTentacle = await new ClientAndTentacleBuilder(tentacleType)
+                .WithTentacleVersion(version)
+                .WithRetriesDisabled()
+                .WithPortForwarderDataLogging()
+                .WithResponseMessageTcpKiller(out var responseMessageTcpKiller)
+                .WithTentacleServiceDecorator(new TentacleServiceDecoratorBuilder()
+                    .LogCallsToFileTransferService()
+                    .CountCallsToFileTransferService(out var fileTransferServiceCallCounts)
+                    .RecordExceptionThrownInFileTransferService(out var fileTransferServiceException)
+                    .DecorateFileTransferServiceWith(b =>
+                    {
+                        b.BeforeUploadFile((service, _, ds) =>
+                        {
+                            service.EnsureTentacleIsConnectedToServer(Logger);
+                            // Only kill the connection the first time, causing the upload
+                            // to succeed - and therefore failing the test - if retries are attempted
+                            if (fileTransferServiceException.UploadLatestException == null)
+                            {
+                                responseMessageTcpKiller.KillConnectionOnNextResponse();
+                            }
+                        });
+                    })
+                    .Build())
+                .Build(CancellationToken);
+
+            var remotePath = Path.Combine(clientTentacle.TemporaryDirectory.DirectoryPath, "UploadFile.txt");
+
+            var uploadFileTask = clientTentacle.TentacleClient.UploadFile(remotePath, DataStream.FromString("Hello"), CancellationToken);
+            
+            Func<Task> action = async () => await uploadFileTask;
+            await action.Should().ThrowAsync<HalibutClientException>();
+            
+            fileTransferServiceException.UploadLatestException.Should().NotBeNull();
+            fileTransferServiceCallCounts.UploadFileCallCountStarted.Should().Be(1);
```

Comment:
Should we check `remotePath` to make sure the file really wasn't uploaded?

Reply:
We can't check that because of the way we're killing the connection 
(unless I'm misunderstanding something). The file will get uploaded, but the _response_ will never arrive.

Reply:
We can't check that, due to the way we kill the connection. We kill the connection when we receive a _response_ from the Tentacle, not when sending the request. Therefore, the file will have been uploaded, but we just don't get the response.

We could add a successful/unsuccessful count to the call count decorators? We currently just count 'started' and 'completed'.